### ISSUE TITLE
(My) Typical case boilerplate reduction

### ISSUE THREAD

**Issue by japgolly** [japgolly/scalajs-react#78]
I normally start a new component like this:
```scala
  val Component =
    ReactComponentB[Props]("Table")
      .stateless
      .backend(new Backend(_))
      .render(_.backend.render)
      .build

  final class Backend($: BackendScope[Props, Unit]) {
    def render: ReactElement = {
      <.table()
    }
  }
```

I could reduce code something like this:
```scala
  def blah[P] = new {
    @inline def apply[B <: {def render: ReactElement}](name: String, f: BackendScope[P, Unit] => B) =
      ReactComponentB[Props]("Table")
        .stateless
        .backend(new Backend(_))
        .render(_.backend.render)
  }
```

Which would let me do:
```scala
  val Component = blah[Props]("Table", new Backend(_)).build
```

Worth it?

**Comment by Kyprus** [comment]
This looks like a big win to me. I do basically the same as you and a built-in helper would be great.

**Comment by japgolly** [comment]
Let's revisit this after #87 is decided. This might be obsolete.

**Comment by japgolly** [comment]
Closed in 63aea4a241f23f9a142e53926a4f5f49d176c7a3

The snippet  at the top of this issue becomes:
```scala
  val Component =
    ReactComponentB[Props]("Table")
      .renderBackend[Backend]
      .build
```
